File: 2025/day6/program1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func main() {

	file, _ := os.Open("input.txt")
	scanner := bufio.NewScanner(file)

	var values [][]int
	var partialAnswers []int

	var finalAnswer int
	var re = regexp.MustCompile(`\s+`)
	for scanner.Scan() {
		line := scanner.Text()
		lineNormalized := strings.TrimSpace(line)
		lineNormalized = re.ReplaceAllString(lineNormalized, " ")
		lineValues := strings.Split(lineNormalized, " ")

		for i, value := range lineValues {
			intValue, err := strconv.Atoi(value)
			if err != nil {
				break
			}
			if i > len(values)-1 {
				values = append(values, []int{})
			}
			values[i] = append(values[i], intValue)
		}

		for i, sign := range lineValues {
			if sign == "*" {
				answer := 1
				for _, value := range values[i] {
					answer *= value
				}
				partialAnswers = append(partialAnswers, answer)
			} else if sign == "+" {
				answer := 0
				for _, value := range values[i] {
					answer += value
				}
				partialAnswers = append(partialAnswers, answer)
			}
		}
		for _, value := range partialAnswers {
			finalAnswer += value
		}
	}
	fmt.Println(values)
	fmt.Println(partialAnswers)
	fmt.Println(finalAnswer)

}

File: 2025/day7/program1.py
from pprint import pprint

lines = [list(line.strip()) for line in open("input.txt")]

counter = 0
for i in range(1, len(lines)):
    for j in range(len(lines[i])):
        if lines[i][j] == ".":
            if lines[i - 1][j] == "S":
                lines[i][j] = "|"
            if lines[i - 1][j] == "|":
                lines[i][j] = "|"

        elif lines[i][j] == "^":
            if lines[i - 1][j] == "|" and j > 0 and j < len(lines[i]) - 1:
                is_split = False
                if lines[i][j - 1] != "^":
                    lines[i][j - 1] = "|"
                    is_split = True
                if lines[i][j + 1] != "^":
                    lines[i][j + 1] = "|"
                    is_split = True
                if is_split:
                    counter += 1

pprint(lines)
print(counter)
